os/arch/arm/src/common/up_createstack.c

```diff
@@ -268,31 +272,26 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 
 #ifdef CONFIG_STACK_COLORATION
 #ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
-		if (uheap) {
-			/* If CONFIG_MPU_STACK_OVERFLOW_PROTECTION is enabled, stack is created with requested size + CONFIG_MPU_STACK_GUARD_SIZE.
-			 * In this case, we should make coloration from stack_alloc_ptr to adj_stack_ptr.
-			 * So we should re-calculate the coloration size with CONFIG_MPU_STACK_GUARD_SIZE.
-			 */
-			up_stack_color(tcb->stack_alloc_ptr, tcb->adj_stack_size + CONFIG_MPU_STACK_GUARD_SIZE);
-		} else
+		/* If CONFIG_MPU_STACK_OVERFLOW_PROTECTION is enabled, stack is created with requested size + CONFIG_MPU_STACK_GUARD_SIZE.
+		 * In this case, we should make coloration from stack_alloc_ptr to adj_stack_ptr.
+		 * So we should re-calculate the coloration size with CONFIG_MPU_STACK_GUARD_SIZE.
+		 */
+		up_stack_color(tcb->stack_alloc_ptr, tcb->adj_stack_size + CONFIG_MPU_STACK_GUARD_SIZE);
+#else
+		/* Currently kernel thread does not have a stack guard at front of stack so that
+		 * the size of coloration should be the tcb->adj_stack_size.
+		 */
+		up_stack_color(tcb->stack_alloc_ptr, tcb->adj_stack_size);
 #endif
-		{
-			/* Currently kernel thread does not have a stack guard at front of stack so that
-			 * the size of coloration should be the tcb->adj_stack_size.
-			 */
-			up_stack_color(tcb->stack_alloc_ptr, tcb->adj_stack_size);
-		}
 #endif
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
 #ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
-		if (uheap) {
-			/* The smallest size that can be programmed for an MPU region is 32 bytes */
-			mpu_get_register_config_value(&tcb->stack_mpu_regs[0], MPU_REG_NUM_STK,
-				(uint32_t)tcb->stack_alloc_ptr, CONFIG_MPU_STACK_GUARD_SIZE, true, false);
-		}
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+		mpu_get_register_config_value(&tcb->stack_mpu_regs[0], MPU_REG_NUM_STK,
+			(uint32_t)tcb->stack_alloc_ptr, CONFIG_MPU_STACK_GUARD_SIZE, true, false);
```

Comment:
@amandeepc ,
Can you change the os/arch/arm/Kconfig details as well to support kernel stack protection.